internal/ethereum/get_receipt.go

```diff
@@ -97,6 +119,75 @@ func ProtocolIDForReceipt(blockNumber, transactionIndex *fftypes.FFBigInt) strin
 	return ""
 }
 
+/*
+Accepts hex string, with or without leading 0x and also accepts short hex (leading zeros removed)
+Converts the hex string to bytes then decodes it as call data for the given abi.Entry
+*/
+func decodeCallDataHex(ctx context.Context, function *abi.Entry, callDataHex string) (*abi.ComponentValue, error) {
+	callDataHex = strings.TrimPrefix(callDataHex, "0x")
+	if len(callDataHex)%2 == 1 {
+		callDataHex = "0" + callDataHex
+	}
+
+	callDataBytes, err := hex.DecodeString(callDataHex)
+	if err != nil {
+		return nil, err
+	}
+
+	componentValue, err := function.DecodeCallDataCtx(ctx, callDataBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return componentValue, nil
+}
+
+func (c *ethConnector) getErrorMessage(ctx context.Context, transactionHash string) (*string, error) {
+
+	var debugTrace *txDebugTrace
+
+	traceErr := c.backend.CallRPC(ctx, &debugTrace, "debug_traceTransaction", transactionHash)
+	if traceErr != nil {
+		return nil, traceErr.Error()
+	}
+
+	returnValue := debugTrace.ReturnValue
+	if returnValue == "" {
+		// some clients (e.g. Besu) include the error reason on the final struct log
+		if len(debugTrace.StructLogs) > 0 {
+			finalStructLog := debugTrace.StructLogs[len(debugTrace.StructLogs)-1]
+			if *finalStructLog.Op == "REVERT" && finalStructLog.Reason != nil {
+				returnValue = *finalStructLog.Reason
+			}
+		}
+	}
+	if returnValue == "" {
+		// not found return value in any of the expected places
+		return nil, nil
+	}
+
+	parsedABI, err := abi.ParseABI([]byte(`[
+			{
+				"name": "Error",
+				"inputs": [
+					{"name": "message", "internalType": "string", "type": "string" }
+				],
+				"type": "function"
+			}
+		]`))
+	if err != nil {
+		return nil, err
+	}
+
+	value, err := decodeCallDataHex(ctx, parsedABI.Functions()["Error"], returnValue)
+	if err != nil {
+		return nil, err
+	}
+	errorMessage := value.Children[0].Value
```

Comment:
I assume we can never get here with `value.Children` having 0 elements because `decodeCallDataHex()` would return an error if the ABI didn't match?